### PR TITLE
Summary table for Section 5, fixes #70

### LINKED ISSUE #70: Summary table for 5.1
Comment from Ekr's review of -10:

5.1.  Pre-Connection Interfaces

   Configuration interfaces are used to configure the security protocols
   before a handshake begins or the keys are negotiated.

This presentation seems pretty hard to follow, in that it's a lot of
work to figure out which in the long list of protocols doesn't have
this property. Maybe a summary table would help?

### PR DESCRIPTION
**Pull request by renghardt**
Summarized the interfaces exposed to applications in a table to make Section 5 easier to read, see #70.

I pretty much used the information as given in Section 5.1 to 5.3 as is, however, I have the following issues/questions:

- Are we intentionally missing gQUIC? Is it the same as (IETF) QUIC?
- Are we intentionally missing ZRTP?
- Some entries list IKEv2, some list ESP - I merged the two, as this document considers them together according to Section 3.4.1. Is this correct here?
- It looks to me like OpenVPN is missing some interfaces. For example, as far as I know, it is possible to provide an identity and private key for OpenVPN, and it is possible to choose which algorithms to support. Should we add these?
- For Pre-Shared Key Import/Export: Is "non-importable" the same as "Interface is not exposed"?
- For Pre-Shared Key Export: Is "TLS (for QUIC)" just "QUIC" and is "DTLS (for SRTP)" just "SRTP"? TLS and DTLS are also given on their own below.
- Do we want to delete the lists of protocols from Sections 5.1 to 5.3 once we have completed the table in Section 5.4?

### CODE REVIEW

**Comment by csperkins** [comment]
> Are we intentionally missing ZRTP?

Not sure – is the table is intended to represent plain SRTP or SRTP with some key management protocols (which could be either DTLS or ZRTP)?

File: draft-ietf-taps-transport-security.md [hunk 1]
```diff
@@ -443,16 +443,19 @@ cannot decouple the security from transport functionality.
...
 pre-connection (configuration), connection, and post-connection interfaces, following
 conventions in {{?I-D.ietf-taps-interface}} and {{?I-D.ietf-taps-arch}}.
 
+Note that not all protocols support each interface.
+The table in {{interface-protocols-table}} summarizes which protocol exposes which of the interfaces.
+
 ## Pre-Connection Interfaces
 
 Configuration interfaces are used to configure the security protocols before a
 handshake begins or the keys are negotiated.
```
Comment (tfpauly) [inline draft-ietf-taps-transport-security.md:453]: It may be good to mention that the abbreviations used here are just keys for the table, not "official" acronyms or anything.

**Review by tfpauly — APPROVED** [review]
Looks good, with one comment!

File: draft-ietf-taps-transport-security.md (hunk 1 above)
Comment (renghardt) [reply draft-ietf-taps-transport-security.md:453]: Added a sentence that explains why we provide abbreviations.
If anyone wants to phrase this in a different way, I'm open to suggestions.

**Comment by renghardt** [comment]
> > Are we intentionally missing ZRTP?
> 
> Not sure – is the table is intended to represent plain SRTP or SRTP with some key management protocols (which could be either DTLS or ZRTP)?

The table is intended to represent exactly the same as the lists of protocols in Sections 5.1 through 5.3.
As these lists only say "SRTP" without mentioning a specific key management protocol, I guess right now it represents SRTP.
If the interfaces are independent of the key management protocol, then this is correct and we are intentionally missing ZRTP. But if there's different interfaces for DTLS-SRTP and ZRTP, then we should separate them out.

File: draft-ietf-taps-transport-security.md [hunk 2]
```diff
@@ -575,6 +579,32 @@ as use of a new Connection Identifier (CID).
...
+The following table summarizes which protocol exposes which interface.
+
+|---
+| Protocol  | IPK | ALG | EXT | CM | AD | PSKI | IV | SAV | CT | KU | PSKE | KE | ME |
+|:----------|:---:|:---:|:---:|:--:|:--:|:----:|:--:|:---:|:--:|:--:|:----:|:--:|:--:|
+| TLS       | x   | x   | x   | x  |    | D    | x  |     | x  | x  | D    |    |    |
+| DTLS      | x   | x   | x   | x  |    | D    | x  | x   | x  | x  | D    |    |    |
+| SRTP      | x   | x   |     |    | x  |      | x  |     |    |    | E    |    |    |
+| QUIC      | x   | x   | x   | x  |    | E    | x  | x   | x  | x  | E    |    | x  |
+| tcpcrypt  |     | x   |     |    |    | D    |    |     | x  | x  | E    |    |    |
```
Comment (tfpauly) [inline draft-ietf-taps-transport-security.md]: Let's update these from #83

**Comment by csperkins** [comment]
@theri DTLS-SRTP and ZRTP will have different interfaces, so we'll need to separate them out. I'll do a PR.

**Comment by renghardt** [comment]
I updated the table from #82 and #83 and double-checked that the table matches the lists.
Agree with keeping both the lists and the table.

@csperkins Sounds good! Feel free to update the lists and then I'll adjust the table.

I think my other questions from the initial post have mostly been answered.

I'll just assume we are intentionally missing gQUIC from the lists and table.

The one question I'm still wondering about is Pre-Shared Import/Export:
For PSK import, the lists say "Non-importable" for CurveCP, while SRTP and IKEv2 are missing from the list. Are they the same, i.e., should we explicitly put SRTP and IKEv2 as "Non-importable"?
Same question for PSK export, where for CurveCP it's "Non-exportable", while ESP and WireGuard are missing from the list.
Please advise.

**Comment by tfpauly** [comment]
I think IKEv2 should be Direct Import for PSKI. Not sure about SRTP.

For PSK export, it doesn't really apply to either ESP or WireGuard, so I think they should be non-exportable.

**Comment by renghardt** [comment]
Updated the table to match #86 (ZRTP interfaces) and resolved merge conflict.
Double-checked that the table matches the lists and #86.
I think this is good to go.
